### PR TITLE
Feature/semantic versioning

### PR DESCRIPTION
**Pull request by matthewhartstonge**
Implements TuneLab/truss#211

As mentioned in #211, this currently needs to be re-worked to use `go dep` instead of glide.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,31 +1,34 @@
 # Makefile for Truss.
 #
-SHA := $(shell git rev-parse --short=10 HEAD)
 
-MAKEFILE_PATH := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
-VERSION_DATE := $(shell $(MAKEFILE_PATH)/commit_date.sh)
+# Pre-install required binaries
+$(shell go get github.com/golang/protobuf/protoc-gen-go)
```
Comment (adamryman) [inline Makefile:5]: Do not do this. This means that whenever the make file is run these will be executed. Even when running tests.

File: Makefile [hunk 2]
```diff
@@ -1,31 +1,34 @@
...
+VERSION_DATE := $(shell go-datefmt -ts ${GIT_COMMIT_EPOC} -fmt UnixDate -utc)
 
 # Build native Truss by default.
 default: truss
 
 dependencies:
-	go get github.com/golang/protobuf/protoc-gen-go
-	go get github.com/golang/protobuf/proto
-	go get github.com/jteeuwen/go-bindata/...
+	glide install
```
Comment (adamryman) [inline Makefile:19]: These are binary dependencies for truss, not dependencies of a truss service.

**Review by adamryman — COMMENTED** [review]
Hi @MatthewHartstonge.

If you want to change the default dependency management, please still commit all of the vendored code.

truss is a binary, and such we want to keep all of our vendored code in our repo, just in case those dependencies disappear.

If you want to make a good argument for removing them, we are up to hear it.

I would still want the git commit in the generated code as well. As we do not want to bump the semantic version for every commit we make, though we still want to know what the sha of the compiled version of truss was that generated code. As this allows us to know exactly what someone has installed.

In addition, please update the CI to work. You can get travis to run the ci on your fork if you want to test things.

Thanks for your interest! We look forward to seeing what you come up with.

**Comment by matthewhartstonge** [comment]
Cheers for the feedback, will add to my todo list. 
For now I'm going to close this PR until I get time to deal with this issue again.

**Pull request closed without merging**